forest/benchmarking/direct_fidelity_estimation.py

```diff
@@ -171,223 +220,107 @@ def _monte_carlo_dfe(program: Program, qubits: Sequence[int], in_states: list, n
+    :param qubits: The qubits to perform DFE on. This can be a superset of the qubits
+        used in ``program``.
+    :param benchmarker: The `BenchmarkConnection` object used to design experiments
+    :param n_terms: Number of randomly chosen observables to measure. This number should be 
+        a constant less than ``2**len(qubits)``, otherwise ``exhaustive_process_dfe`` is more efficient.
     """
-    pauli_est = data.expectation / np.abs(cal.expectation)
-    var_est = ratio_variance(data.expectation, data.variance, cal.expectation, cal.variance)
-
-    temp_data = { 'dimension': data.dimension, 'expectation': pauli_est, 'variance': var_est }
-
-    if type == 'state':
-        fid = aggregate_state_dfe(temp_data)
-    elif type == 'process':
-        fid = aggregate_process_dfe(temp_data)
-    else:
-        raise ValueError('Error: must specify state or process DFE.')
-
-    return DFEestimate(
-        in_pauli=data.in_pauli,
-        program=data.program,
-        out_pauli=data.out_pauli,
-        dimension=data.dimension,
-        number_qubits=data.number_qubits,
-        pauli_point_est=pauli_est ,
-        pauli_var_est=var_est,
-        fid_point_est=fid['expectation'],
-        fid_var_est=fid['variance'],
-        fid_std_err_est=np.sqrt(fid['variance'])
-    )
-
-
-def aggregate_process_dfe(data: dict):
-    if isinstance(data['dimension'], int):
-        dim = data['dimension']
+    exp = TomographyExperiment(list(
+        _monte_carlo_dfe(program=program, qubits=qubits,
+                         in_states=[None, plusX, minusX, plusY, minusY, plusZ, minusZ],
+                         n_terms=n_terms, benchmarker=benchmarker)),
+        program=program, qubits=qubits)
+    return DFEexperiment(exp,'process')
+
+
+def acquire_dfe_data(qc: QuantumComputer, dfe: DFEexperiment, n_shots = 10_000, active_reset=False):
+    # TODO: add number of shots
+    #       add variance bounds?
+    res = list(measure_observables(qc, dfe.exp, n_shots=n_shots, active_reset=active_reset))
+    return DFEdata(res, 
+                   in_states = [str(exp[0].in_state) for exp in dfe.exp],
+                   program = dfe.exp.program,
+                   out_pauli = [str(exp[0].out_operator) for exp in dfe.exp], 
+                   pauli_point_est = np.array([r.expectation for r in res ]), 
+                   pauli_var_est = np.array([r.stddev**2 for r in res]), 
+                   cal_point_est = np.array([r.calibration_expectation for r in res ]), 
+                   cal_var_est = np.array([r.calibration_stddev**2 for r in res ]), 
+                   dimension = 2**len(dfe.exp.qubits), 
+                   qubits = dfe.exp.qubits,
+                   type = dfe.type)
+
+
+def analyse_dfe_data(data: DFEdata):
+    mean = np.sum([r.expectation for r in data.res])
+    variance = np.sum([r.stddev**2 for r in data.res])
+    d = data.dimension
+
+    if data.type == 'state':
+        mean_est = np.mean([r.expectation for r in data.res])
```

Comment:
Again, is this the same as mean/len(data.res) or data.pauli_point_est/len(data.res)

same for var_est

Reply:
👍